openhexa/toolbox/dhis2/api.py

```diff
@@ -40,7 +38,13 @@ def __init__(self, connection: DHIS2Connection, cache_dir: Optional[Union[Path,
         self.session.mount("https://", adapter)
         self.session.mount("http://", adapter)
 
-        self.session = self.authenticate(connection.username, connection.password)
+        if connection is None and ("url" not in kwargs and "username" not in kwargs and "password" not in kwargs):
```

Comment:
No, you need either a connection object , or all three kwargs to configure the client (url, username, password)

Reply:
I think that your condition is incorrect since you use `and` between the kwargs

You state that we need :
`connection OR (url AND username AND password)`
The negation of this is :
`NOT connection AND (NOT url OR NOT username OR NOT password)`

Let me know if I miss something

Reply:
No, you dont miss it is correct